### ISSUE TITLE
People: Update pattern for success message

### ISSUE THREAD

**Issue by sixhours** [Automattic/wp-calypso#32229]
**Steps to reproduce:** 
Update information for a user.

**What we see:**

The success message appears in a banner message that is relative to the card you're editing. 

- Should this banner be full screen?
- Does it make sense for this message to be a banner or should it appear in a snackbar?
- Would using a snackbar be accessible?

**Comment by alaczek** [comment]
Could you clarify what action did you perform to get this banner? Did you remove a follower?

**Comment by sixhours** [comment]
It looks like this message appears after you edit a teammate:

<img width="757" alt="Screen Shot 2019-04-12 at 11 51 18 AM" src="https://user-images.githubusercontent.com/2124984/56050255-60234300-5d19-11e9-851a-223080e9e80a.png">

And it doesn't disappear when you switch to another view (like Followers). I think that's the confusion.

**Comment by alaczek** [comment]
I tried different actions within People section:
  * sending an invitation, 
  * accepting invitation as a user
  * accepting invitation to follow
  * deleting an invitation
In all those cases the success message appears in the snackbar. Removing the user seems to be the only exception, where the success notification is displayed in full width above the card. If we want to make things consistent we'd have to move this message to snackbar.

> Would using a snackbar be accessible?
I didn't dive into this one, but do you have any specific accessibility concerns here? If so, I'm assuming it would be a larger issue to address outside of this ticket, since snackbar messages are used all over Calypso.

One last related thing I noticed while testing - when removing a follower there's no success message. Do you think it warrants a separate issue?

**Comment by sixhours** [comment]
I flagged this for the Muriel team (added label `Muriel`) since I think it was originally raised by Courtney, and because they are the snack bar people. :D

**Comment by scalarbane2** [comment]
![Desktop 2 1](https://user-images.githubusercontent.com/50745054/58037026-5c5abb80-7b24-11e9-81ce-ad49b8969ab1.png)
![Desktop 2](https://user-images.githubusercontent.com/50745054/58037027-5cf35200-7b24-11e9-902f-334417cd6351.png)

I tried using Snackbars to represent the message and this is what it looks like.

**Comment by cburton4** [comment]
Hey y'all,

This is a good first pass 👍A few things I see:

The green used doesn't look consistent with our [color palette](https://automattic.github.io/color-studio/index.html). I can't be 100% sure but it looks like the color used in the screenshot of the live UI is Green 600.

The guidance for snackbars says they should be placed consistently in the bottom left hand corner of the screen for wide screen layouts. That said though we have had several accessibility conversations around why this might not be the best pattern for everyone. @enriquesanchez would love to hear your thoughts and guidance here.

**Comment by scalarbane2** [comment]
Hello!
I updated the color with exactly the one in the UI components in calypso. 
![Edit User - Sucess](https://user-images.githubusercontent.com/50745054/58089210-236e2580-7bbc-11e9-88d0-555e1ad53882.png)
![Remove Follower - Success](https://user-images.githubusercontent.com/50745054/58089211-236e2580-7bbc-11e9-88e7-78e89e15fd27.png)

I also tried fixing the snackbar on the bottom left of the screen. In my opinion, it might not be visible to all the users. Meanwhile, I was thinking if it's possible for the snackbar to display relative to where the changes were made. For example, if the user makes a change towards the bottom of the screen, the snackbar should appear at the bottom left and vice-versa. What do you think?
![Remove Follower - Success-1](https://user-images.githubusercontent.com/50745054/58089581-100f8a00-7bbd-11e9-9588-3298418d5ecb.png)

**Comment by enriquesanchez** [comment]
> would love to hear your thoughts and guidance here.

Thanks @cburton4 for bringing this to my attention. 

> The pattern for the snackbar would be the correct pattern to use in this instance as the information is optional (although very useful) and would disappear automatically 

Agreed, because of how they are displayed and announced to assistive technology, snackbars should be used only for optional or "nice to know" information. 

It is also recommended to leave enough time for automatically disappearing snackbars so users who need more time to process things on the screen can actually see them. One time limit I've seen recommended in other places is around 10s and it' also important to note that messages should be short and clear.

It's important that the snackbar is properly announced to assistive technology, so use of `role: status` and `aria-live: polite` is required.

As for positioning, have we tried having it displayed centered? While snackbars are not ideal for users with low vision (particularly those with low peripheral vision), I think having appear in the bottom-center of the screen might help with that.

**Comment by scalarbane2** [comment]
![Edit User - Sucess](https://user-images.githubusercontent.com/50745054/58190418-627fa200-7cb4-11e9-9461-a7a1f3e3364c.png)
![Remove Follower - Success](https://user-images.githubusercontent.com/50745054/58190419-63183880-7cb4-11e9-9689-35af2e1668c8.png)

@cburton4 and @sixhours Let me know what you think.

**Comment by drw158** [comment]
I prefer the snackbar to be located bottom center as well, although this preference is not based on a11y or testing. Because we have many competing elements on the edges of screens (nav drawer, top app bar, etc.) bottom center makes the most sense to me.

Note this would be a component change and would impact all of Calypso. It'd be a pretty big change that would need to be tested.

**Comment by alaczek** [comment]
> Note this would be a component change and would impact all of Calypso. It'd be a pretty big change that would need to be tested.

Yes! I agree this goes beyond this ticket and requires more thorough review as to the impact throughout Calypso. So maybe to bring this particular notification in line with others, we move it to top right, and address the component change in separate issue?

**Comment by scalarbane2** [comment]
Hello! So I tried taking the hover tip to the top and also explored another type of solution.

![On Hover](https://user-images.githubusercontent.com/50745054/58825337-29d9b400-8636-11e9-90ad-9edfa6985221.png)

For this second one, I brought the error below the field. I personally don't feel right about this, but let me know what you think. 

![On Hover-1](https://user-images.githubusercontent.com/50745054/58825409-4ece2700-8636-11e9-8752-2864e2591ab7.png)

**Comment by sixhours** [comment]
@scalarbane2 Looks like that update is for a different ticket, #32397 -- they're similar so it's an easy mistake to make. :D

**Comment by sixhours** [comment]
So it sounds like, going forward, we'll want to open a PR to move the success message into a traditional snackbar for the People section, and moving the snackbar position itself will be addressed in #33441.

I'm moving this back to To Do now that we have mock-ups for this issue, so someone from dev can take a look!
